addons/markers/functions/fnc_onButtonClickConfirm.sqf

```diff
@@ -17,12 +17,28 @@
 params ["_buttonOk"];
 
 private _display = ctrlParent _buttonOk;
-private _description = _display displayctrl IDC_INSERT_MARKER;
+private _description = _display displayCtrl IDC_INSERT_MARKER;
 private _aceTimestamp = _display displayCtrl IDC_ACE_INSERT_MARKER_TIMESTAMP;
 
 // handle timestamp
 if (cbChecked _aceTimestamp && {ACE_player call FUNC(canTimestamp)}) then {
-    private _time = daytime;
+    private _time = switch (GVAR(timestampTimezone)) do {
+        case 1: {
+            systemTime params ["", "", "", "_hour", "_min", "_sec"];
+            _hour + _min/60 + _sec/3600
+        };
+        case 2: {
```

Comment:
Add comment explaining why it's done this way

Reply:
Sorry you mean the switch statement as a whole or just the part you selected?

Reply:
> Add comment explaining why it's done this way

Added comments to the switch. If you mean that specific part why the first 3 params are not included and the _min and _sec are devided, i'm not really sure how I would comment that without being to technical or wordy.

Reply:
Sorry, I meant case 2 specifically. It's fine if it's wordy, but that needs to be there lest an onlooker reverts it in the future.

Reply:
Oke I think this makes sense.

```sqf
case 2: {
    systemTimeUTC params ["", "", "", "_hour", "_min", "_sec"];
    _hourOffset = round (GVAR(timestampUTCOffset));
    _hour = _hour + _hourOffset;

    // add or subtract minutes offset based on the negative or positive timezone
    _min = if (_hourOffset < 0) then { _min - GVAR(timestampUTCMinutesOffset) } else { _min + GVAR(timestampUTCMinutesOffset) };

    // prevent the timestamp from exceeding 24 hours or going below 0 hours
    _time = ((_hour + _min/60 + _sec/3600) % 24 + 24) % 24;
    _time
};
```